lib/manager/gomod/artifacts.ts

```diff
@@ -17,19 +16,18 @@ import type {
   UpdateArtifactsResult,
 } from '../types';
 
-function getPreCommands(): string[] | null {
-  const credentials = find({
-    hostType: PLATFORM_TYPE_GITHUB,
-    url: 'https://api.github.com/',
-  });
-  let preCommands = null;
-  if (credentials?.token) {
-    const token = quote(credentials.token);
-    preCommands = [
-      `git config --global url.\"https://${token}@github.com/\".insteadOf \"https://github.com/\"`, // eslint-disable-line no-useless-escape
-    ];
+function getGoEnvironmentVariables(registryUrls: string[]): NodeJS.ProcessEnv {
+  let goEnvVariables: NodeJS.ProcessEnv = {};
+
+  for (const registryUrl of registryUrls) {
+    const goPrivateWithProtocol = getHttpUrl(`https://${registryUrl}`);
```

Comment:
```suggestion
    const goPrivateWithProtocol = getHttpUrl(registryUrl);
```

`registryUrl` should / is already be full url